src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -408,11 +408,6 @@ protected void paintIcon(final Graphics g, final AbstractButton b, final Rectang
             }
         } else if (model.isPressed() && model.isArmed()) {
             tmpIcon = b.getPressedIcon();
-            if (tmpIcon == null) {
-                if (icon instanceof ImageIcon) {
-                    tmpIcon = new ImageIcon(AquaUtils.generateSelectedDarkImage(((ImageIcon)icon).getImage()));
```

Comment:
Any other reason to remove the usage of grayimage as the pressedIcon other than this failed test? Probably it is better to set the pressedIcon in the test?